src/docs/admin/references/NotificationServiceReference.adoc

```diff
@@ -295,3 +295,204 @@ and if they have a value depending on the event origin.
+----
+
+All the relevant properties are located at the end of the file. We will detail them now.
+
+----
+notification.script.path=dist/war/notification-service/WEB-INF/classes/microsoftTeamsMessage.groovy
+# URL of the script, it will be processed if notification.script.path is empty
+notification.script.url=
+#optional parameter for specifying Script Engine
+notification.script.engine=groovy
+#if false execute the script one time per event even if there are several subscribers
+#if true execute the script for each subscribers for each event
+notification.script.multiple.execution=true
+----
+
+- *notification.script.path*: The notification service can execute a single Third-Party script, for all users. This property specifies the script that will be executed.
+    It can't be dynamically updated and requires a server restart to be changed.
+- *notification.script.url:* Allows specifying the script url instead of using a path. Not used here
+- *notification.script.engine:* Specifies the engine used to execute the script. Here it must be set to groovy as we are executing a groovy script.
+- *notification.script.multiple.execution:* Sets the execution mode to single or multiple execution. Explained <<execution-mode,here>>
+    We will be using this script in multiple execution mode as we want each user to be able to send notifications using their microsoft accounts on different channels
+
+Once the properties are updated as the code snippet above, you can save the file.
+
+===== Configuring Microsoft 365
+Because the script queries the Microsoft Graph API (a REST API) to interact with microsoft Teams, it needs to connect to Microsoft as a user (retrieve an authentication token) through a registered app.
+
+This App is either already present or we can create a dedicated one for the ProActive Notification Service. +
+Steps on how to create the App are present in the following description.
+
+The Notification's Third-Party script needs 2 IDs and a secret, that are retrieved from the Microsoft Entra admin center:
+
+- A *tenantId* -> The ID of your organisation
+- A *clientId* -> The ID of the registered App
+- A *secret* -> The "password", encoded using a ProActive tool, used to authenticate the app when querying the Graph API
+
+====== Connect to the Microsoft entra center
+
+- Connect to your Microsoft 365 account from https://account.microsoft.com/account/Account[here]
+- Go to "My Apps" from the left menu
+
+image::microsoft-my-apps-button.png[align="center"]
+
+- Access the Admin center by clicking on the Admin app
+
+image::microsoft-apps-dashboard-admin.png[align="center"]
+
+- Access the Microsoft Entra admin center by clicking on "Identity" in the "Administration centers" left menu
+//TODO get screenshot image::microsoft-apps-dashboard-Entra-admin-button.png[align="center"]
+- Click on the "App Registration" sub menu in the "Applications" left menu
+
+image::entra-admin-center.png[align="center"]
+
+====== OPTIONAL: Create an App
```

Comment:
I think we should not declare this as optional, you need some specific settings in the app that any random existing app will not necessarily have.

Reply:
OK